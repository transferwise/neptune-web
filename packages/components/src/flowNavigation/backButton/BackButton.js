import React from 'react';
import Types from 'prop-types';
import classNames from 'classnames';
import { ArrowLeft as ArrowLeftIcon } from '@transferwise/icons';

import './BackButton.css';

const BackButton = ({ label, className, onClick }) => (
  <button
    type="button"
    className={classNames('np-back-button', 'align-items-center', 'btn-unstyled', className)}
    onClick={onClick}
  >
    <ArrowLeftIcon size={24} />
    {label}
  </button>
);

BackButton.propTypes = {
  className: Types.string,
  label: Types.element,
  onClick: Types.func.isRequired,
};

BackButton.defaultProps = {
  className: undefined,
  label: undefined,
};

export default BackButton;
