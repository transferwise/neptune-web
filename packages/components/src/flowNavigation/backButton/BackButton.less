.np-back-button {
  color: var(--color-control-accent);
  display: flex !important;

  &:hover {
    color: var(--color-control-accent-hover);
  }
}
