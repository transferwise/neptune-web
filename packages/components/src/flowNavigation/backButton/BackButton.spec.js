import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

import BackButton from '.';

const props = {
  label: <>label</>,
  className: 'className',
  onClick: jest.fn(),
};
describe('BackButton', () => {
  it(`renders as expected`, () => {
    expect(render(<BackButton {...props} />).container).toMatchSnapshot();
  });
});
