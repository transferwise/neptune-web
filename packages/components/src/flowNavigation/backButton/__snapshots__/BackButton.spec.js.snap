// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton renders as expected 1`] = `
<div>
  <button
    class="np-back-button align-items-center btn-unstyled className"
    type="button"
  >
    <span
      aria-hidden="true"
      class="tw-icon tw-icon-arrow-left "
      role="presentation"
    >
      <svg
        fill="currentColor"
        focusable="false"
        height="24"
        width="24"
      >
        <path
          d="M5.814 13H22v-2H5.814l6.893-6.893L11.293 2.7 2 12l9.293 9.3 1.414-1.407L5.814 13z"
        />
      </svg>
    </span>
    label
  </button>
</div>
`;
